badge_server/main.py

```diff
@@ -137,24 +138,53 @@ def get_highest_status(cls, statuses: Iterable[enum.Enum]):
 }
 
 
-def _get_supported_versions(package_name: str) -> List[int]:
-    """Gets the given package's supported python versions
+def _get_missing_details(package_names: List[str],
+                         results: Iterable[CompatibilityResult]) -> str:
+    """Gets the details for any missing data (if there is any)
 
     Args:
-        package_name: The name of the package to look up, e.g. "tensorflow".
+        package_names: A list of length 1 or 2 of the package name(s) to look
+            up, e.g. ["tensorflow"], ["tensorflow", "opencensus"].
+        results: A list of length 1 or 2 of the `CompatibilityResults` for the
+            given package(s).
 
     Returns:
-        A list of supported python versions.
+        None if there is no missing data; a description of the version(s)
+        missing otherwise, e.g. "Missing data for python version(s): 2".
     """
-    supported = []
-    if not compat_utils._is_package_in_whitelist([package_name]):
-        return supported
+    expected_lengths = (1, 2)
+    bad_list_length = 'package_names: Expected length of 1 or 2, got %s'
+    not_whitelisted = 'One of the packages in %s is not whitelisted'
+
+    if len(package_names) not in expected_lengths:
+        # This should not happen
+        raise ValueError(bad_list_length % len(package_names))
+
+    if not compat_utils._is_package_in_whitelist(package_names):
+        # This should not happen
+        raise ValueError(not_whitelisted % str(package_names))
+
+    versions = (2, 3)
     unsupported_package_mapping = configs.PKG_PY_VERSION_NOT_SUPPORTED
-    for version in (2, 3):
-        unsupported_packages = unsupported_package_mapping[version]
-        if package_name not in unsupported_packages:
-            supported.append(version)
-    return supported
+    expected = {version: all([name not in unsupported_package_mapping[version]
```

Comment:
Maybe this is more readable?

```python
package_supported_versions = {2, 3}
for version in [2, 3]:
  for package_name in package_names:
    if package_name in configs.PKG_PY_VERSION_NOT_SUPPORTED[version]:
      package_supported_versions.discard(version)

data_for_versions = {result.python_major_version for result in results}

if data_for_versions >= package_supported_versions:
  return None
else:
  return 'Missing compatibility for packages={}, versions={}'.format(package_names, package_supported_versions - data_for_versions)
```

Reply:
Done, but I made some minor tweaks to the variable names and used `issuperset()` instead of `>=` since that seemed clearer to me.